src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionTextCache.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ExpressionTextCache
+    {
+        private readonly ConcurrentDictionary<LambdaExpression, string> _cacheEntries  = 
+                new ConcurrentDictionary<LambdaExpression, string>();
+
+        public string Get(LambdaExpression expression)
+        {
+            if (expression == null)
+            {
+                throw new ArgumentNullException(nameof(expression));
+            }
```

Comment:
If this is `.Internal`, can assume calling code has already done the `null` checking. Feel free to `Debug.Assert()` that.